launcher/minecraft/auth/AccountList.cpp

```diff
@@ -122,8 +122,17 @@ void AccountList::addAccount(const MinecraftAccountPtr account)
 
     // override/replace existing account with the same profileId
     auto profileId = account->profileId();
-    if(profileId.size()) {
-        auto existingAccount = findAccountByProfileId(profileId);
+    if(profileId.size() && account->isMojangOrMSA()) {
+        int existingAccount = -1;
+        for (int i = 0; i < count(); i++) {
+            MinecraftAccountPtr existing = at(i);
+            if (existing->profileId() == profileId &&
+                existing->isMojangOrMSA()) {
+                existingAccount = i;
+                break;
+            }
+        }
```

Comment:
Sure, I've replaced this with a `std::find_if`. Might as well use it for new code. It might be more readable, though C++ does its best to make that not the case :)

Reply:
oh, thank you! I got a little confused